### PR TITLE
Categorize papers by recency

### LINKED ISSUE #27: Categorize article collection by date
The article date stamps `2022-10-31` are redundant and hard to parse. Also we want to emphasize the time/recency, in general.
Categorize the articles so its easier to understand how new an article is

Missed this in a previous conversation (https://github.com/PathwayCommons/hyper-recent/discussions/17)

### PR DESCRIPTION
**Pull request by Suhyma**
Filters through all recent papers from the past month and creates the following categories/headings in the UI:
- Today's papers
- Papers from the past 3 days
- Papers from the past week
- Rest of the papers from this month

### CODE REVIEW

File: src/dashboard/category-results-screen.js [hunk 1]
```diff
@@ -15,8 +16,62 @@ function Paper ({ paper }) {
...
+function findRelativeDate (papers, range) {
+  const now = new Date();
+  const today = format(now, 'yyyy-MM-dd');
+  const displayPapers = [];
+
+  if (range === 'today') {
+    for (const paper of papers) {
+      const paperDate = format(paper.date, 'yyyy-MM-dd');
+      if (paperDate === today) {
+        displayPapers.push(paper);
```
Comment (Suhyma) [inline src/dashboard/category-results-screen.js:28]: At the moment, all the categories seem to work as expected except for today's papers. When the data has papers from the day of, the `displayPapers` array here is still returned as empty. I'm currently looking into this, however, I've been having some issues with debugging since these functions are only run when the app is rendered in `app.js`. I've put up what I have at the moment in `launch.json` for some feedback, but perhaps I should be adding these date filtering functions in a new JS file? Not too sure, let me know your thoughts!

**Comment by jvwong** [comment]
Cross-reference #27

File: src/dashboard/category-results-screen.js (hunk 1 above)
Comment (jvwong) [reply src/dashboard/category-results-screen.js:28]: See comment for  src/dashboard/categories.js

File: src/dashboard/category-results-screen.js [hunk 2]
```diff
@@ -15,8 +16,62 @@ function Paper ({ paper }) {
   ]);
 }
 
+function findRelativeDate (papers, range) {
+  const now = new Date();
+  const today = format(now, 'yyyy-MM-dd');
+  const displayPapers = [];
+
+  if (range === 'today') {
```
Comment (jvwong) [inline src/dashboard/category-results-screen.js:24]: Maybe single loop and use https://date-fns.org/v2.29.3/docs/isWithinInterval

File: src/dashboard/category-results-screen.js [hunk 3]
```diff
@@ -15,8 +16,62 @@ function Paper ({ paper }) {
...
+        displayPapers.push(paper);
+      }
+    }
+  }
+  return displayPapers;
+}
+
 export default function CategoryResultsScreen ({ store }) {
   const { selectedPapers, selectedCategory } = store;
+  const todayPapers = findRelativeDate(selectedPapers, 'today');
```
Comment (jvwong) [inline src/dashboard/category-results-screen.js:71]: What if one or more of these is empty?

**Review by jvwong — CHANGES_REQUESTED** [review]

File: src/dashboard/category-results-screen.js [hunk 4]
```diff
@@ -15,19 +16,73 @@ function Paper ({ paper }) {
...
+  const weekAgo = sub(now, { weeks: 1 }).toISOString();
+  const monthAgo = sub(now, { months: 1 }).toISOString();
+  const displayPapers = [];
+
+  for (const paper of papers) {
+    const paperDate = paper.date.toISOString().split('T')[0];
+
+    if (range === 'today' && paperDate === today) { // Today's papers
+      displayPapers.push(paper);
+    } else if (range === 'days' && paperDate >= daysAgo && paperDate < today) { // Papers from 4 days ago
```
Comment (Suhyma) [inline src/dashboard/category-results-screen.js:32]: I attempted to use isWithinInterval here, but soon realized it had a lot of complications with timezones - the downloaded data would be in UTC, but it would get automatically converted to EST when comparing it to the current date which made things a bit more difficult to work with. I found that it was easier to manipulate date strings rather than Date objects for our purposes.

**Review by jvwong — APPROVED** [review]
